Added a document of the deprecating process (#7672)
cri-o/cri-o: deprecating_process.md

> <p>Config options are the settings that allow us to control various aspects of the behaviour and functionality of CRIO. They allow us to customise how CRI-O operates within a kubernetes cluster.</p>

Reviewer: Can this have a title?

# General Deprecating Process of CRI-O

> **What are config options?**

Reviewer: I would use `###` in place of the bold `**`

Config options are the settings that allow us to control various aspects of the behaviour and functionality of CRIO. They allow us to customise how CRI-O operates within a kubernetes cluster.

**Types of config options:**

1)General options: These control overarching settings like logging levels, storage locations, and state directories.

2)Image options: These manage image storage, pulling, and default volumes.

3)Network options: These configure container networking, including plugins and configuration directories.

4)Runtime options: These govern the container runtime itself, such as the OCI runtime binary, pause command, and container monitor configuration.

> 5)Security options: These enforce security measures like AppArmor and seccomp profiles.

Reviewer: personally I don't know if we need this section

**General process of deprecating a config option:**

Config options undergo deprecation only during major and minor version changes in CRI-O. Removals do not occur abruptly within patch releases, ensuring a smoother transition for users adapting to evolving configurations.

A deprecated status is assigned to a configuration option for a minimum of one release cycle before it is actually removed. This gives users a warning and time to adjust their configurations to adapt to the upcoming changes.

The deprecation is communicated through multiple channels, including updates in the documentation, messages to indicate deprecated config option in the CRI-O CLI help text and a log entry within CRI-O itself.

Configuration options are typically removed if they are no longer considered necessary or if they have been replaced by alternatives, especially options present in the Kubelet.

> Several CRI-O config options have been replaced by Kubelet as the management of runtime containers has shifted towards the Kubelet.</p>

Reviewer: no need for </p> if below you use `###` instead of `**`

**Examples of crio.conf options getting replaced by Kubelet:**

1)Runtime Management was previously done with the “runtime\_path” config option which was replaced with the “--container-runtime” flag in Kubelet.

2)Pod PIDs Limit was previously done with the “pids\_limit” config option which was replaced with the “--pods\_pids-limit” flag in Kubelet

> 

Reviewer: can you replace the `1)` `2)` `3)` with `- `

# General Deprecating Process of CRI-O

## What are config options?

> Config options are the settings that allow us to control various aspects of the behaviour and functionality of CRIO. They allow us to customise how CRI-O operates within a kubernetes cluster.

Reviewer 2: I'm not a fan of pronouns when writing documents. Something like this is more professional and more apparent, in my opinion.

```suggestion
Config options are the settings that can be used to control various aspects of the behavior and functionality of CRIO. They allow customization of CRI-O operations within a Kubernetes cluster.
```

At the very least, capitalize Kubernetes, and then consider using the American English version of "behavior" vs "behaviour". I'd also suggest changing "config" to "configuration" everywhere.

Author: This is my first contribution to open source, still getting the hang of it. I will make it more formal.
